SwiftyTesseract/SwiftyTesseract/SwiftyTesseract.swift

```diff
@@ -186,7 +190,64 @@ public class SwiftyTesseract {
     completionHandler(swiftString)
     
   }
-
+  
+  /// Takes an array UIImages and returns the PDF as a data object
+  ///
+  /// - Parameter images: Array of UIImages that should be analyzed
+  /// - Returns: PDF as Data object
+  /// - Throws: SwiftyTesseractError in case something went wrong
+  public func createPDF(from images: [UIImage]) throws -> Data {
+    let _ = semaphore.wait(timeout: .distantFuture)
+    
+    // create unique file path
+    let file = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(NSUUID().uuidString)
+    
+    guard let renderer = TessPDFRendererCreate(file.path, bundle.pathToTrainedData, 0) else {
+        throw SwiftyTesseractError.unableToCreateRenderer
+    }
+    
+    guard TessResultRendererBeginDocument(renderer, "Unkown Title") == 1 else {
+      TessDeleteResultRenderer(renderer)
+      throw SwiftyTesseractError.unableToBeginDocument
+    }
+    
+    var pixImages = [Pix]()
+    defer {
+      // Release the Pix instance from memory
+      for var pix in pixImages {
+        pixDestroy(&pix)
+      }
+      semaphore.signal()
+    }
+    
+    for page in 0..<images.count {
```

Comment:
Following the same logic as above, instead of using a range to iterate over the images, you can now iterate over `pixImages`:
```suggestion
try pixImages.enumerated().forEach { [weak self] pageNumber, pix in
    guard let self = self else { return }
    guard TessBaseAPIProcessPage(self.tesseract, pix, Int32(pageNumber), "page.\(pageNumber)", nil, 0, renderer) == 1 else { 
        throw SwiftyTesseractError.unableToProcessPage
    }